saos-persistence/src/main/java/pl/edu/icm/saos/persistence/common/IndexableObject.java

```diff
@@ -5,30 +5,49 @@
 
 import org.joda.time.DateTime;
 
+/**
+ * Abstract entity class representing persistence data object
+ * that can be indexed
+ * 
+ * @author madryk
+ */
 @MappedSuperclass
 public abstract class IndexableObject extends DataObject {
 
     private boolean indexed = false;
     
     private DateTime indexedDate;
-    
-    public void markAsIndexed() {
-        indexed = true;
-        indexedDate = new DateTime();
-    }
-    
-    public void resetIndexedFlag() {
-        indexed = false;
-    }
 
+
+    /**
+     * Is entity needs to be indexed.
```

Comment:
Returns false if the given object tree current state has not been indexed (and should be indexed)